src/loader/mod.rs

```diff
@@ -150,12 +95,26 @@ impl Display for Error {
     }
 }
 
-#[derive(Debug, Default, Copy, Clone, PartialEq)]
-/// Result of the KernelLoader load() call.
+#[cfg(all(feature = "elf", any(target_arch = "x86", target_arch = "x86_64")))]
+impl From<ElfError> for Error {
+    fn from(err: ElfError) -> Self {
+        Error::Elf(err)
+    }
+}
+
+#[cfg(all(feature = "bzimage", any(target_arch = "x86", target_arch = "x86_64")))]
+impl From<BzImageError> for Error {
+    fn from(err: BzImageError) -> Self {
+        Error::Bzimage(err)
+    }
+}
+
+/// Result of [`KernelLoader.load()`](trait.KernelLoader.html#tymethod.load).
 ///
 /// This specifies where the kernel is loading and passes additional
 /// information for the rest of the boot process to be completed by
 /// the VMM.
+#[derive(Default)]
```

Comment:
Can we keep all the derives? They are actually pretty nice when debugging the code.

Reply:
Done